andru1989/vowels.rb

```diff
@@ -37,26 +40,57 @@ def read_file
     print_vowels
   end
 
+  # Public
+  #
+  #   => Set initial accessor with Time.now
+  #   => Get the content of 'text.txt' file and loop each slice with(20000)
+  #   => Acumulate in '@threads' Array a new Thread
+  #   => For each ary it call the 'set_values' method to update de Hash
+  #   => Wait a moment while each Thread finish
+  #   => Set final accessor with Time.now
+  #   => Then it call the 'print_vowels' method to print the hash
+  def read_thread_file
+    self.initial = Time.now
+    open_file.each_slice(20000) do |ary|
+      @threads << Thread.new do
+        set_values(ary)
+      end
+    end
+    @threads.each{|t| t.join}
+    self.final = Time.now
+
+    print_vowels
+  end
+
   private
   # Private:
   #
   #   => Print the Hash @vowels
   #   => Print the difference between initial and final accessors to show how many seconds the process long
+  #   => Set 0 for each key of the hash
   def print_vowels
     p @vowels
     p "Run in #{(self.final - self.initial)} Seconds"
+    @vowels.keys.each{|k| k = 0}
   end
 
   # Private:
   #
   #   => Do a loop of the '@vowels' keys
-  #   => Scan in the current line after convert it to downcase if the current key(vowel) was found
+  #   => Validate if the value is 'String' class
+  #   => Scan in the current line after convert it to downcase if the current key(vowel) was found and the class is 'String'
+  #   => If the class of the value isn't string first do a 'join' and then a downcase to scan it
   #   => Accumulate in the current key the size of the Array that the scan method returned
   #
   # Parameters:
-  #   => It receives the current line of the opened file
-  def set_values(line)
-    @vowels.keys.each{|k| @vowels[k] += line.downcase.scan(k).length}
+  #   => It receives the current 'String' or 'Array' of the opened file
+  def set_values(value)
+    @vowels.keys.each do |k|
+      @vowels[k] += case value.class.to_s
+        when 'String' ; value.downcase.scan(k).length
```

Comment:
Indent `when` as deep as `case`.<br>Space found before semicolon.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.